### ISSUE TITLE
 ERROR  Cannot set property 'requiresBuild' of undefined

### ISSUE THREAD

**Issue by pgonzal** [pnpm/pnpm#1217]
### pnpm version: 2.x

I have a repository that is failing `pnpm install` with this error:

```
Packages: +1525
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Resolving: total 3426, reused 3421, downloaded 5, done
 ERROR  Cannot set property 'requiresBuild' of undefined
at <anonymous>  …node_modules/supi/lib/link/index.js:163
at next
at fulfilled    …b/node_modules/supi/lib/link/index.js:4
```

### Code to reproduce the issue:

1. I start with a clean clone of the repo, and delete the `shrinkwrap.yaml` file.
2. I run `pnpm install` using PNPM 1.43.1 it installs fine, and creates a shrinkwrap file
3. Then I start over with PNPM 2.2.2 and I get the above error.  
4. Then I start over with PNPM 2.4.0.  With this version, the PNPM statistics show that it is installing twice as many packages (3426 instead of 1583), and it prints the same error but with a slightly different line number:

```
Packages: +1525
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Resolving: total 3426, reused 3421, downloaded 5, done
 ERROR  Cannot set property 'requiresBuild' of undefined
at <anonymous>  …node_modules/supi/lib/link/index.js:163
at next
at fulfilled    …b/node_modules/supi/lib/link/index.js:4
```

This is a private repo so I can't share it on GitHub.  When I get some time I can collect verbose logs and email them to you, and/or provide the package.json files.

### Additional information:

 - `node -v` prints: 8.10.0
 - Windows, OS X, or Linux?: Windows

**Comment by pgonzal** [comment]
@zkochan I emailed you an archive with a repro, in case that helps.

**Comment by zkochan** [comment]
Thanks for sending the project for repro. I was able to reproduce with a smaller `package.json`:

```json
{
  "dependencies": {
    "@microsoft/gulp-core-build-webpack": "3.4.5",
    "webpack": "~3.11.0"
  }
}

```

**Comment by zkochan** [comment]
The crash fixed in 2.6.0

**Comment by pgonzal** [comment]
Thanks very much for getting this fixed so quickly! :-)

**Comment by pgonzal** [comment]
There are two separate bugs here.  The first one is fixed (thanks!), so to avoid confusion I am closing this GitHub issue. I opened issue #1233 to track the incorrect install statistics.

**Comment by vjpr** [comment]
@zkochan Just encountered this bug again:

```
 ERROR  Cannot set property 'requiresBuild' of undefined
at <anonymous>  /Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/pnpm/2.17.5/node_modules/pnpm/lib/node_modules/supi/lib/install/link.js:241
at next
at fulfilled    /Users/Vaughan/nvm/versions/node/v8.9.0/pnpm-global/1/node_modules/.registry.npmjs.org/pnpm/2.17.5/node_modules/pnpm/lib/node_modules/supi/lib/install/link.js:4
```

**Comment by zkochan** [comment]
@vjpr I published your fix with 2.17.7
